test/test_sync.cpp

```diff
@@ -3678,27 +3624,35 @@ TEST(Sync_UploadDownloadProgress_6)
     client_config.one_connection_per_session = false;
     Client client(client_config);
 
+    util::ScopeExit cleanup([&]() noexcept {
+        client.shutdown_and_wait();
+        server.stop();
+        server_thread.join();
+    });
+
     Session::Config session_config;
     session_config.server_address = "localhost";
     session_config.server_port = server_port;
     session_config.realm_identifier = "/test";
+    session_config.service_identifier = "/realm-sync";
     session_config.signed_user_token = g_signed_test_user_token;
 
     std::mutex mutex;
     std::condition_variable session_cv;
+    bool signaled = false;
     auto session = std::make_unique<Session>(client, db, nullptr, nullptr, std::move(session_config));
 
     auto progress_handler = [&](uint_fast64_t downloaded_bytes, uint_fast64_t downloadable_bytes,
                                 uint_fast64_t uploaded_bytes, uint_fast64_t uploadable_bytes,
-                                uint_fast64_t progress_version, uint_fast64_t snapshot_version) {
+                                uint_fast64_t snapshot_version, double, double) {
         CHECK_EQUAL(downloaded_bytes, 0);
         CHECK_EQUAL(downloadable_bytes, 0);
         CHECK_EQUAL(uploaded_bytes, 0);
         CHECK_EQUAL(uploadable_bytes, 0);
-        CHECK_EQUAL(progress_version, 0);
-        CHECK_EQUAL(snapshot_version, 1);
+        CHECK_EQUAL(snapshot_version, 3);
```

Comment:
why did this change? the setup seems the same.

Reply:
first progress call with `progress_version == 1` was with the `snapshot_version == 3`, so it's functionally the same checks adapted on `m_reliable_download_progress` refactoring